File: 01_projects/code/java-playwright/cbaps-dawms/robust/src/main/java/com/playwright/dawms/pages/SubmissionIntakePage.java
package com.playwright.dawms.pages;
import com.microsoft.playwright.*;
import com.playwright.cbaps.library.EnhancedPlaywrightManager;
import com.playwright.dawms.models.SubmissionData;
import org.slf4j.*;

public class SubmissionIntakePage {
    private Page page;
    private EnhancedPlaywrightManager pwm;
    private Locator submissionTypeDropdown, applicationNumberInput, createSubmissionButton;
    
    public SubmissionIntakePage(Page page, EnhancedPlaywrightManager pwm) {
        this.page = page; this.pwm = pwm;
        this.submissionTypeDropdown = page.locator("#submissionType");
        this.applicationNumberInput = page.locator("#applicationNumber");
        this.createSubmissionButton = page.locator("button:has-text('Create Submission')");
        pwm.waitUntilElementVisible("#submissionType");
    }
    
    public ReviewerAssignmentPage createSubmission(SubmissionData data) {
        pwm.selectDropdown(submissionTypeDropdown, data.getSubmissionType());
        pwm.enterText(applicationNumberInput, data.getApplicationNumber());
        pwm.clickElement(createSubmissionButton);
        return new ReviewerAssignmentPage(page, pwm);
    }
}

File: 01_projects/code/python-playwright/cbaps-dawms/robust/dawms/pages/submission_intake_page.py
from playwright.sync_api import Page
from loguru import logger

class SubmissionIntakePage:
    def __init__(self, page: Page, pw):
        self.page = page
        self.pw = pw
        self.type_dropdown = page.locator("#submissionType")
        self.app_number_input = page.locator("#applicationNumber")
        self.create_btn = page.locator("button:has-text('Create Submission')")
        self.pw.wait_for_selector("#submissionType")
    
    def create_submission(self, data):
        logger.info(f"📝 Creating submission: {data.application_number}")
        self.pw.select_dropdown(self.type_dropdown, data.submission_type)
        self.pw.enter_text(self.app_number_input, data.application_number)
        self.pw.click_element(self.create_btn)
        from .reviewer_assignment_page import ReviewerAssignmentPage
        return ReviewerAssignmentPage(self.page, self.pw)
